syntax = "proto3";

package productservice;

option go_package = "./productservice";

// Definições de mensagens
message Product {
  int32 id = 1;
  string name = 2;
  double price = 3;
  string description = 4;
  string image = 5;
}

message StandardResponse {
  bool success = 1;
  string message = 2;
}

message GetProductsRequest {}

message GetProductsResponse {
  repeated Product products = 1;
}

message GetProductByIdRequest {
  int32 id = 1;
}

message AddProductRequest {
  Product product = 1;
}

message UpdateProductRequest {
  Product product = 1;
}

message DeleteProductRequest {
  int32 id = 1;
}

message HealthCheckResponse {
  string status = 1;
  string version = 2;
}

// Definição do serviço
service ProductService {
  rpc GetHealth(GetProductsRequest) returns (HealthCheckResponse);
  rpc GetProducts(GetProductsRequest) returns (GetProductsResponse);
  rpc GetProductById(GetProductByIdRequest) returns (Product);
  rpc AddProduct(AddProductRequest) returns (StandardResponse);
  rpc UpdateProduct(UpdateProductRequest) returns (Product);
  rpc DeleteProduct(DeleteProductRequest) returns (StandardResponse);
}
